### PR TITLE
store: requst another PD node if some of them is down

### LINKED ISSUE #26303: query from `information_schema.tikv_region_status` will fail if specific pd-server node down
When a query tries to fetch data from the `information_schema.tikv_region_status`, tidb-server will try to get the address of a pd-server node, and request from this node using http api.

https://github.com/pingcap/tidb/blob/d193cb2c36c4dab6b91ce07a3c3b31f56ccd1845/executor/infoschema_reader.go#L1383-L1392
https://github.com/pingcap/tidb/blob/d193cb2c36c4dab6b91ce07a3c3b31f56ccd1845/store/helper/helper.go#L753-L757

https://github.com/pingcap/tidb/blob/d193cb2c36c4dab6b91ce07a3c3b31f56ccd1845/store/helper/helper.go#L767-L780

The pd-server node is chosen is always the `pdHosts[0]`. When a pd-server node is marked down but not removed from the cluster, the address of this node will still be returned from the `pdClient`. This will cause the problem that tidb-server still tries to request from the pd-server node even if it is marked as down. Thus query may fail:
``` sql
> select region_id, group_concat(store_id order by store_id), group_concat(right(label ->> '$[0].value',1) order by label) from tikv_region_status join tikv_region_peers using (region_id) join tikv_store_status using (store_id) where db_name='test' and index_name='PRIMARY' group by region_id order by region_id;
ERROR 1105 (HY000): Get "http://127.0.0.1:2382/pd/api/v1/regions": dial tcp 127.0.0.1:2382: connect: connection refused
```

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve?

Issue Number: close #26303  

Problem Summary:
As the problem described in the issue to be fixed.

### What is changed and how it works?

What's Changed:
Request another PD node if some of them is down.

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- Fix the problem that tidb-server tries to request from a PD node even if it's down when quering the table in information_schema database.

### CODE REVIEW

File: store/helper/helper.go [hunk 1]
```diff
@@ -862,9 +876,16 @@ func (h *Helper) GetStoresStat() (*StoresStat, error) {
...
+	req := new(http.Request)
+	for _, host := range pdHosts {
+		req, err = http.NewRequest("GET", util.InternalHTTPSchema()+"://"+host+pdapi.Stores, nil)
+		if err != nil {
+			// Try to request from another PD node when some nodes may down.
+			if strings.Contains(err.Error(), "connection refused") {
+				continue
+			}
+			return nil, errors.Trace(err)
+		}
```
Comment (crazycs520) [inline store/helper/helper.go:888]: How about extract a function to avoid duplicate code?

**Review by crazycs520 — APPROVED** [review]
LGTM

**Review by jyz0309 — APPROVED** [review]

**Review by zimulala — APPROVED** [review]
LGTM

**Comment by lhy1024** [comment]
it also needs to cherry-pick other versions, such as 5.0.x

**Pull request merged**